### DISCUSSION TITLE
"invalid header: 0x0a324655" flashing micropython onto esp32s3

### DISCUSSION THREAD

**Discussion by noahcoad** [micropython/micropython#16158] [ESP32] [answered]
I have a "esp32s3 Super Mini" and a "ESP32 S3 Development Board" that I'm not able to get Micropython running on.

I'm able to erase with: `esptool.py --chip esp32s3 --port $PORT erase_flash`	 
and then flash with: `esptool.py --chip esp32s3 --port $PORT write_flash -z 0 <firmware>`
but any of the esp32s3 firmwares I try (generic, generic 4MB, generic oxtal, tinyS3, nanoS3, omgS3, etc) result in "**invalid header: 0x0a324655**" continually repeated when I connect to the serial port using picocom.

Any ideas?

I've tried with and without -z for compression.

The only reference to that address I've found is on this uf2 tool page https://github.com/microsoft/uf2 it says the file offset 0 size 4 bytes should be "First magic number, 0x0A324655", which I assume indicates it's a UF2 file.

![esp32s3 Super Mini](https://m.media-amazon.com/images/I/71l6PM1-CJL._AC_SL1500_.jpg) from https://www.amazon.com/dp/B0D47HBFDY

![ESP32 S3 Development Board](https://m.media-amazon.com/images/I/71FrYaHAlKL._AC_SL1500_.jpg) from https://www.amazon.com/dp/B0DB1WK3CW

Flashing...
```
esptool.py v4.8.1
Serial port /dev/cu.usbmodem1101
Connecting....
Detecting chip type... ESP32-S3
Chip is ESP32-S3 (QFN56) (revision v0.2)
Features: WiFi, BLE, Embedded Flash 4MB (XMC), Embedded PSRAM 2MB (AP_3v3)
Crystal is 40MHz
MAC: f0:f5:bd:79:22:84
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Flash will be erased from 0x00000000 to 0x00309fff...
Compressed 3186688 bytes to 1157876...
Writing at 0x0010b510... (30 %)
```

After flashing on the serial port...
```
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40049b1e
invalid header: 0x0a324655
invalid header: 0x0a324655
invalid header: 0x0a324655
invalid header: 0x0a324655
invalid header: 0x0a324655
invalid header: 0x0a324655
invalid header: 0x0a324655
invalid header: 0x0a324655
invalid header: 0x0a324655
invalid header: 0x0a324655
```

**Comment by glenn20** [comment, marked as answer]
Are you using the ".bin" firmware files (not .uf2)?

**Reply by noahcoad** [reply]
# 🤦🏼‍♂️
omg. that was it. using the .bin worked.
wow, surprised I missed that. thank you!
